pkgs/development/libraries/reactphysics3d/default.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "reactphysics3d";
+  version = "0.9.0";
+
+  src = fetchFromGitHub {
+    owner = "DanielChappuis";
+    repo = "reactphysics3d";
+    rev = "v${finalAttrs.version}";
```

Comment:
I am generally not a friend of this.

This does not help overriding the version of a package, just confuse users.

Either move the version into a `let`, hardcode it in both locations or make it part of the `src`-drvs `name` (which is sadly discouraged).

Reply:
In the latter case, there would be *some* error reporting, on overrding the version rather than just substituting the old sources (as it would happen now).

At the end, whatever you do, an explicit override of the `src` will always be required, and therefore using `finalAttrs` in the definition of `src` doesn't make sense.

Reply:
we should just use rec here

Reply:
Unfortunate, I'm the one who told them to use finalAttrs. As far as I know, it'd the preferred pattern

Reply:
Will change back to rec after work